docker/common/install_cuda_libs.sh

```
@@ -48,32 +48,38 @@ install_ubuntu_requirements() {
-        libcublas${CUBLAS_MAJOR_VER}-dev-cuda-${CUDA_MAJOR_VER}=${CUBLAS_VER} \
-        cuda-nvrtc-dev-${NVRTC_CUDA_VERSION}=${NVRTC_VER}
+    # Skip remove+reinstall for any library already at the target version (e.g. pre-installed
+    # by the base image at a version not yet published to the public CUDA apt repo).
+    installed_pkg_version() {
+        dpkg-query -W -f='${Version}' "$1" 2>/dev/null || true
+    }
+
+    PKGS_TO_INSTALL=()
+    if [[ "$(installed_pkg_version libcudnn9-cuda-13)" != "${CUDNN_VER}" ]]; then
+        apt-get remove --purge -y libcudnn9* || true
+        PKGS_TO_INSTALL+=(libcudnn9-cuda-13=${CUDNN_VER} libcudnn9-dev-cuda-13=${CUDNN_VER} libcudnn9-headers-cuda-13=${CUDNN_VER})
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)